siphon/catalog.py

```diff
@@ -369,6 +369,10 @@ def __init__(self, element_node, catalog_url=''):
 
         """
         self.name = element_node.attrib['name']
+        if 'ID' in element_node.attrib:
+            self.id = element_node.attrib['ID']
+        else:
+            self.id = self.name
```

Comment:
I tend to agree with @lesserwhirls here. I think setting to `None` is probably the best default behavior.